src/window.c

```diff
@@ -315,3 +468,15 @@ char *time_to_string(double current_time, double duration)
    return time_string ;
 }
 
+int get_number_of_tracks(GSList *list)
+{
+   int i = 1 ;
+   while (list->next != NULL)
+   {
+      ++i ;
+      list = list->next ;
+   }
+   return i ;
+}
```

Comment:
consider using `g_slist_length (list) ;` here. Also note that your function returns 1 even is there is an empty list.

Reply:
Oh neat function now using `g_slist_length (list) ;` here.